### PR TITLE
Add hints for "install and set up mongoose"

### PR DESCRIPTION
**Pull request by ailyntang**
* Add hint for how to solve the timeout error

* Add hint for what to put in the `.env` file

* Add code snippets for `.env`, `package.json` and `myApp.js`

- [x ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x ] My pull request targets the `master` branch of freeCodeCamp.
- [ x] None of my changes are plagiarized from another source without proper attribution.
- [x ] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x ] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@ailyntang 

Keep the hints separate from the solution code.  The following is part of the solution which is already at the bottom.  Just remove the following changes:

>Once you remove the comma, the tests will run correctly. The correct code:
```
"dependencies": {
  "express": "^4.12 .4",
  "body-parser": "^1.15.2"
},
```
Other than making sure actual solution code is not outside the Solutions section, I think the overall content changes are good. 

We are restructuring all of these hint/solutions.  Only use headers where shown below.  For other sections in your article, just use bolding (surround text with `**`) Please use following structure.  If a section is not needed, do not include the header for it.

````
# Challenge Title

## Problem Explanation
This section gives a summary of what is needed to be done without just copying/pasting the instructions.  Ideally, this header would be placed above lines 6-14.

## Hints

### Hint #1

### Hint#2

#### Relevant Links
These would be relevant Links for the entire Hints section above

## Solutions

<details><summary>Solution #1 (Click to Show/Hide)</summary>

```javascript
// code goes here
```

#### Code Explanation
* Use bullets for the various steps explanation if you want to have anything for Code Explanation

#### Relevant Links
* Use bullets for each link you want to use for this particular solution
</details>

<details><summary>Solution #2 (Click to Show/Hide)</summary>

```javascript
// code goes here
```

#### Code Explanation
* Use bullets for the various steps explanation if you want to have anything for Code Explanation

#### Relevant Links
* Use bullets for each link you want to use for this particular solution
</details>

````

**Comment by ailyntang** [comment]
Thanks @RandellDawson , I've made the changes. 

By the way, the content on this hints page is not showing up on the website. Instead the website is showing outdate content: https://guide.freecodecamp.org/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/

Let me know if you'd like me to raise an issue for this. I assume this is already a known issue.

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 1]
```diff
@@ -12,6 +12,108 @@ OR
 2. Once done add the MONGO_URL in .env file and save your path as 
 ```` mongodb://<dbuser>:<dbpassword>@ds<PORT>.mlab.com:<PORT>/<DATABASE-NAME> ```` which you copy from mLab. Remember to remove the angle brackets ````<  >```` when you replace your username and password of your database.
 
+## Hints
 
+### Hint #1: Timeout error
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
### Hint #1
**Timeout error**
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 2]
```diff
@@ -12,6 +12,108 @@ OR
...
+
+For example, this will result in a timeout error:
+```
+"dependencies": {
+  "express": "^4.12.4",
+  "body-parser": "^1.15.2", 
+},
+```
+
+### Hint #2: add MONGO_URI to .env
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
### Hint #2
**add MONGO_URI to .env**
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 3]
```diff
@@ -12,6 +12,108 @@ OR
...
+
+* Remove spaces before and after `=`. This is correct: `MONGO_URI=mongodb...`. This is incorrect: `MONGO_URI = mongodb...`
+
+* Do you have symbols or special characters in your password, e.g. `$&(@`? If so, you will need to translate these into unicode. MongoDB has instructions on how to do this. I would suggest changing your password to be lettes and numbers only for simplicity.
+
+
+## Solutions
+See below for the correct code to make the tests pass.
+
+<details>
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
<details><summary>Solution #1 (Click to Show/Hide)</summary>
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 4]
```diff
@@ -12,6 +12,108 @@ OR
...
+  <summary>.env</summary>
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
**.env**
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 5]
```diff
@@ -12,6 +12,108 @@ OR
...
+  
+```GLITCH_DEBUGGER=true
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
````suggestion
```
GLITCH_DEBUGGER=true
````

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 6]
```diff
@@ -12,6 +12,108 @@ OR
...
+# store your secrets and config variables in here
+# only invited collaborators will be able to see your .env values
+
+# reference these in your code with process.env.SECRET
+SECRET=
+MADE_WITH=
+MONGO_URI=mongodb+srv://ailyntang:<ENTERYOURPASSWORDHERE>@fcccluster-vlas9.mongodb.net/test?retryWrites=true
+# note: .env is a shell file so there can't be spaces around =
+```
+</details>
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 7]
```diff
@@ -12,6 +12,108 @@ OR
...
+
+<details>
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 8]
```diff
@@ -12,6 +12,108 @@ OR
...
+  <summary>package.json</summary>
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
**package.json**
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 9]
```diff
@@ -12,6 +12,108 @@ OR
...
+	},
+	"keywords": [
+		"node",
+		"hyperdev",
+		"express"
+	],
+	"license": "MIT"
+}
+```
+ </details>
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 10]
```diff
@@ -12,6 +12,108 @@ OR
...
+
+<details>
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 11]
```diff
@@ -12,6 +12,108 @@ OR
...
+  <summary>myApp.js</summary>
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
**myApp.js**
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 12]
```diff
@@ -12,6 +12,108 @@ OR
...
+  
+  ```/**********************************************
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
````suggestion
```javascript
/**********************************************
````

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 13]
```diff
@@ -12,6 +12,108 @@ OR
...
+// as `MONGO_URI`. Connect to the database using `mongoose.connect(<Your URI>)`
+const mongoose = require('mongoose');
+mongoose.connect(process.env.MONGO_URI);
+
+/** # SCHEMAS and MODELS #
+/*  ====================== */
+```
+</details>
+  
 <!-- The article goes here, in GitHub-flavored Markdown. Feel free to add YouTube videos, images, and CodePen/JSBin embeds  -->
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
```

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 14]
```diff
@@ -12,6 +12,108 @@ OR
...
+MADE_WITH=
+MONGO_URI=mongodb+srv://ailyntang:<ENTERYOURPASSWORDHERE>@fcccluster-vlas9.mongodb.net/test?retryWrites=true
+# note: .env is a shell file so there can't be spaces around =
+```
+</details>
+
+<details>
+  <summary>package.json</summary>
+  
+  ```{
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
````suggestion
```json
{
````

File: guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md [hunk 15]
```diff
@@ -12,6 +12,108 @@ OR
...
+
+Still having issues? Check the below hints:
+
+* Remove spaces before and after `=`. This is correct: `MONGO_URI=mongodb...`. This is incorrect: `MONGO_URI = mongodb...`
+
+* Do you have symbols or special characters in your password, e.g. `$&(@`? If so, you will need to translate these into unicode. MongoDB has instructions on how to do this. I would suggest changing your password to be lettes and numbers only for simplicity.
+
+
+## Solutions
+See below for the correct code to make the tests pass.
```
Comment (RandellDawson) [inline guide/english/certifications/apis-and-microservices/mongodb-and-mongoose/install-and-set-up-mongoose/index.md]:
```suggestion
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
In addition to the changes requested here, please use 2 space indentation for all JavaScript and JSON code.

Thank you.

**Review by RandellDawson — APPROVED** [review]

**Pull request merged**